src/python/pants/engine/intrinsics.py

```diff
@@ -102,8 +108,13 @@ async def add_prefix(add_prefix: AddPrefix) -> Digest:
 
 @rule
 async def execute_process(
-    process: Process, process_execution_environment: ProcessExecutionEnvironment
+    process: Process,
+    process_execution_environment: ProcessExecutionEnvironment,
+    env_vars: SubprocessEnvironmentVars,
```

Comment:
Your rule needs to request a `SubprocessEnvironment.EnvironmentAware` I believe. That is the trick by which different environments can set different env vars (and more generally is the trick by which different environments can have different values for a subsystem).

Reply:
This doesn't work, because I have to Get(EnvironmentVariables) inside to get the actual values, and it still fails with a missing rule. https://github.com/pantsbuild/pants/pull/21501/commits/26f69bf31bbdbc099edd7f7947e47a022191eb8b

Reply:
Ah, sorry, yes, you were doing the right thing to begin with, I read `SubprocessEnvironmentVars` as `SubprocessEnvironment`.

Reply:
The `get_subprocess_environment` rule takes care of going from `SubprocessEnvironment.EnvironmentAware` to actual dict of env vars.

Reply:
I'll take a closer look in a bit

Reply:
Oh boy, yes, this is more complicated than I first appreciated. 

However, the complication is justified, and adding a new option won't solve it, unless the new option is not EnvironmentAware, which seems simply incorrect.

I will noodle on this a bit more.